merlin-server/constraints-dsl-compiler/src/libs/constraints-edsl-fluent-api.ts

```diff
@@ -1,87 +1,324 @@
   }
 
-  private static new(astNode: AST.Constraint): Constraint {
-    return new Constraint(astNode);
+  public static ForbiddenActivityOverlap(activityType1: string, activityType2: string): Constraint {
+    return new Constraint({
+      kind: AST.NodeKind.ForbiddenActivityOverlap,
+      activityType1,
+      activityType2
+    })
   }
 
-  public static ViolationsOf(expression: WindowsExpression): ViolationsOf {
-    return Constraint.new({
-      kind: AST.NodeKind.ViolationsOf,
-      expression: expression.__astNode,
-    });
+  public static ForEachActivity(activityType: string, alias: string, constraint: Constraint): Constraint {
+    return new Constraint({
+      kind: AST.NodeKind.ForEachActivity,
+      activityType,
+      alias,
+      expression: constraint.__astNode
+    })
   }
 }
 
-interface True extends WindowsExpression {}
-
-export class WindowsExpression {
+export class Windows {
   /** Internal AST node */
   public readonly __astNode: AST.WindowsExpression;
 
-  private constructor(expression: AST.WindowsExpression) {
+  public constructor(expression: AST.WindowsExpression) {
     this.__astNode = expression;
   }
 
-  private static new(expression: AST.WindowsExpression): WindowsExpression {
-    return new WindowsExpression(expression);
+  public static During(alias: string): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionDuring,
+      alias
+    })
+  }
+  public static StartOf(alias: string): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionStartOf,
+      alias
+    })
+  }
+  public static EndOf(alias: string): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionEndOf,
+      alias
+    })
   }
 
-  public static True(): True {
-    return WindowsExpression.new({
-      kind: AST.NodeKind.WindowsExpressionTrue
-    });
+  public if(condition: Windows): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionIfThen,
+      condition: condition.__astNode,
+      expression: this.__astNode
+    })
   }
 
-  public and(...others: WindowsExpression[]): WindowsExpression {
-    return WindowsExpression.new({
+  public and(...others: Windows[]): Windows {
+    return new Windows({
       kind: AST.NodeKind.WindowsExpressionAnd,
       expressions: [
         this.__astNode,
         ...others.map(other => other.__astNode),
       ],
     });
   }
-
-  public or(...others: WindowsExpression[]): WindowsExpression {
-    return WindowsExpression.new({
+  public or(...others: Windows[]): Windows {
+    return new Windows({
       kind: AST.NodeKind.WindowsExpressionOr,
       expressions: [
         this.__astNode,
         ...others.map(other => other.__astNode),
       ],
     });
   }
+  public not(): Windows {
+    return new Windows({
+      kind: AST.NodeKind.WindowsExpressionNot,
+      expression: this.__astNode
+    })
+  }
+
+  public violations(nominal: boolean = true): Constraint {
```

Comment:
this nominal parameter appears unused

Reply:
Removed. Was an idea to allow the user to make `false` a violation, but I decided it wasn't worth since `.not()` is available.